### DISCUSSION TITLE
Some Feature Requests

### DISCUSSION THREAD

**Discussion by STROUDICOUS** [olvidalo/obsidian-chord-sheets#8] [Ideas]
Hi. Really appreciate the plugin!!

The transpose buttons are a remarkable feature I didn't know I needed.

Some simple rhythm notation would be a great addition; using a pipe "|" for barlines and simple "/" for slash notation. e.g. 
|Gmin / / / | C / Dmin / |

Could you add a "Respell Chords" button? e.g. Transposing up 3 semitones from C major would be better as Eb major rather than D# major.

I noticed that altered chords don't render. e.g. C7b13 or G#b9#11. They would be great, even if the instrument chord symbols (i.e. Guitar, Uke, Mandolin) didn't use them or ignored the upper extension.

On that, an option to use no instrument-specific chord symbols would also be good. 

Chord symbol font. I think there are some free one's around. You could put in an option to use a specific music font for the chords. 

Thanks again

**Comment by olvidalo** [comment]
Hi, thank you for your comments!

> Some simple rhythm notation would be a great addition; using a pipe "|" for barlines and simple "/" for slash notation. e.g.
|Gmin / / / | C / Dmin / |

I could have the plugin ignore non-word characters (|, / etc) when determining if a line is a chord line, so that chords would still be highlighted if this kind of rhythm notation is used. Would this be what you're looking for?

> Could you add a "Respell Chords" button? e.g. Transposing up 3 semitones from C major would be better as Eb major rather than D# major.

This has been implemented in @MarcelJulian's pull request #2 and I am currently refining the ui on the [feature/enharmonic-toggle](https://github.com/olvidalo/obsidian-chord-sheets/tree/feature/enharmonic-toggle) branch. No ETA yet but it will be coming!

> I noticed that altered chords don't render. e.g. C7b13 or G#b9#11. They would be great, even if the instrument chord symbols (i.e. Guitar, Uke, Mandolin) didn't use them or ignored the upper extension.

Please feel free to open an issue with any missing chord types. You can have a look at #4 to see what information I need to add these chords (at least for highlighting). Right now, C7b13 is recognised (though no chord diagram is available). G#b9#11 is not (G# with flat 9th and sharp 11th?). It would be helpful if you could do a quick google search for each missing chord type that you'll report to see if it is actually a common notation :)

> On that, an option to use no instrument-specific chord symbols would also be good.

Could you clarify what you mean with instrument-specific chord symbols? Do you mean the chord diagrams for guitar etc.? Those can be disabled in the plugin settings, just disable the chord overview and the chord diagrams on hover. Or am I misunderstanding something?

> Chord symbol font. I think there are some free one's around. You could put in an option to use a specific music font for the chords.

I'd like to keep the rendering as plain text for consistency, modifying rendering and spacing as little as possible. So I won't be implementing a chord symbol font for now. Sorry!

**Reply by olvidalo** [reply]
> Some simple rhythm notation would be a great addition; using a pipe "|" for barlines and simple "/" for slash notation. e.g.
> |Gmin / / / | C / Dmin / |

Rhythm markers are supported in the latest release.

**Comment by drdguidi** [comment]
I would love to see custom chords supported!

The Markdown Chords plugin supports the following syntax:
[Dm7|x57565] or [G13|x,10,x,12,12,12]

The chords name are rendered as Dm7 or G13, but the chords diagram uses the supplied information. 
This works great for custom chords and I can't see a way to achieve this in Chord Sheets yet.

**Reply by stopnasa** [reply]
2nd this!
